hardware/P1MeterBase.cpp

```diff
@@ -159,26 +159,29 @@ bool P1MeterBase::MatchLine()
 				l_exclmarkfound=1;
 				found=1;
 			}
-			else 
+			else
 				continue;
 			break;
 		default:
 			continue;
 		} //switch
-		
+
 		if(!found)
 			continue;
 
 		if (l_exclmarkfound) {
 			time_t atime=mytime(NULL);
-			sDecodeRXMessage(this, (const unsigned char *)&m_p1power, "Power", 255);
 			bool bSend2Shared=(difftime(atime,m_lastSharedSendElectra)>59);
-			if (std::abs(double(m_lastelectrausage)-double(m_p1power.usagecurrent))>40)
-				bSend2Shared=true;
+			if (
+				(std::abs(double(m_lastelectrausage)-double(m_p1power.usagecurrent))>40)&&
+				(difftime(atime,m_lastSharedSendElectra)>29)
```

Comment:
I don't think this is the way to solve it.
A line above we check if the last-send was 60 seconds ago, if yes, send it
Maybe this line should be changed

if difference>40 AND lastsend was atleast 60 seconds ago, then send

But instead of 60 seconds, 10 seconds would be OK as other meters sends data every 10 seconds.
If the data is the same, then because of the value '40' check, it is not send

Other then this, you data is still processed every second
The only thing not done is write the values in the database (and some extra logic)

If we talk about CPU usage, how much does this increase ?